kratos/solving_strategies/convergence_accelerators/convergence_accelerator.h

```diff
@@ -173,6 +191,25 @@ class ConvergenceAccelerator
     // Level of echo for the convergence accelerator
     int mEchoLevel;
 
+    ///@}
+    ///@name Protected  Access
+    ///@{
+
+    virtual void ComputeJacobianApproximation()
+    {
+        KRATOS_ERROR << "Interface Jacobian approximation cannot be computed with this convergence accelerator." << std::endl;
+    }
+
+    virtual DenseMatrixPointerType pGetInverseJacobianApproximation()
+    {
+        KRATOS_ERROR << "Interface Jacobian approximation is not available for this convergence accelerator." << std::endl;
+    }
+
+    virtual unsigned int GetProblemSize()
```

Comment:
why adding those fcts if they are protected?

also why dense? :sweat_smile:

Reply:
Because we will need them to solve the Interface Block Quasi-Newton (IBQN) equations but we don't want to give access to "anyone". The idea is that only the IBQN accelerators can access this data.

This Jacobian is always dense and we need it to be dense in order to solve the interface quasi-Newton problem with a QR decomposition (rather than inverting the matrix).